### ISSUE TITLE
feat: upload and download config

### ISSUE THREAD

**Issue by OwenTruong** [OwenTruong/civitdl#28]

**Comment by aa956** [comment]
something like wget or curl `--limit-rate=7M` would be nice to have if you mean this.

Also not sure if this is download config feature or separate feature but it would be nice to check if the file is already downloaded?
Not sure how to implement or how much shall be checked (e.g. current expected save path vs scan of the entire download directory with hashing) but if I recall correctly now civitdl just redownloads everything in batchfile.

It may be a surprise if batchfile contains some SDXL checkpoints and was (re)started just because of the wrong selection from terminal history :)

```console

Now downloading "Juggernaut XL"...
            - Model ID: 133005
            - Version ID: 240840

 12%|████████▋                                                                  | 823M/7.11G 
```

**Comment by OwenTruong** [comment]
Hi I have added issue #37 and #38 for these. For rate limit, it shouldn't be too bad to implement that.

I agree that it would be better to not have to re-download if possible. 

There are a few ways I am thinking:
- We could scan a root directory for hashes.
- We could scan a root directory for matching filenames.
- We could add a JSON somewhere that would track what models have been downloaded already at a root directory.

Hashes would be the best solution, with JSON being the worst. If it is possible to download hashes from CivitAI, that would be the best. We probably do not want to create hashes locally as that can be slow on some machines.

**Comment by OwenTruong** [comment]
Merged into #29
